dask/dataframe/core.py

```diff
@@ -780,6 +780,45 @@ def min(self, axis=None, skipna=True):
             return self._aca_agg(token='min', func=_min,
                                  skipna=skipna, axis=axis)
 
+    def _idxmaxmin(self, fn, axis=None, skipna=True):
+        if fn == 'idxmax':
+            idxfn = _idxmax
+        elif fn == 'idxmin':
+            idxfn = _idxmin
+        else:
+            raise ValueError('Function must be idxmax or idxmin')
+
+        axis = self._validate_axis(axis)
+        if axis == 1:
+            return map_partitions(idxfn, None, self,
+                                  token=self._token_prefix + fn,
+                                  skipna=skipna, axis=axis)
+        else:
+            def chunk(x):
+                idx = getattr(x, fn)()
+                value = x.iloc[idx]
+                n = len(x)
+                return pd.DataFrame([[idx, value, n]], columns=['idx','value', 'n'])
+
+            def agg(x, **kwargs):
+                print(x)
+                idx = getattr(x.value, fn)()
+                if self.known_divisions == True:
+                    return x.idx.iloc[idx]
+                else:
+                    return x.n.iloc[:idx-1].sum() + x.idx.iloc[idx]
```

Comment:
Two requests for these functions:

1.  Define them outside of this function.  Dynamically generated functions are harder to serialize and are often the cause of surprising slowdowns
2.  Give then informative names like `idxminmax_chunk` and `idxminmax_agg`.  These names will show up in diagnostic tools like the progressbars and so are important.  We need to get better at doing this through the dask.dataframe codebase.

This will probably require that you send the `fn` variable as a keyword argument to the function, which is probably a good idea regardless (closures can be hard to debug).